### PR TITLE
Nutshell: Log in and Registration

### PR DESCRIPTION
**Pull request by Keartroth**
### Comments
- Added a log in form and registration form.
- Added a user provider module to access and distribute the users resource.
- Added a session storage module to act as a user authentication and log in storage.

### Things to Check
- Check that all previous site functionality still works: events, articles, and messages.
- Check that a current user in the users resource in database.json can log on:
 --- In order to check,  serve database.json on local port 8088 with a watch flag (-w) and 
use a fake account previously saved (user name: Steve Brownlee    password: coach).
- Check that a new user can be entered to database.json and that a new user is not entered without entering all required information:
--- In order to check, serve database.json on local port 8088 with a watch flag (-w) and click the "Register" button and fill out all the required fields:
--- If all fields are correctly filled out, and the "Sign Up" button is clicked, an alert box will appear with the following message, "Welcome, user name" with the entered user name.
- Check that passwords must match in order to create a user account:
--- An alert should state, "Re-entered password does not match initial password." if passwords do not match.

### CODE REVIEW

**Review by durrjp — APPROVED** [review]
Looks great and all components working as they should

**Review by iparkerhodge — APPROVED** [review]
👍 everything works for me as well. nicely done.

File: src/scripts/login/SignInForm.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        <div id="registerButtonContainer"></div>
+        <div id="usernameContainer">
+            <label class="label label--login" for="loginUserName">Username:</label>
+            <input type="text" id="loginUserName" required>
+        </div>
+        <div id="passwordContainer">
+            <label class="label label--login" for="loginPassword">Password:</label>
+            <input type="text" id="loginPassword" required>
+        </div>
+        <div id="loginButtonContainer"></div>
```
Comment (stevebrownlee) [inline src/scripts/login/SignInForm.js:19]:
```js
${LoginButton()}
```

File: src/scripts/login/SignInForm.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+* Author: Michael Carroll
+* This module holds an HTML representation of the sign in form. The function, SignInForm,
+* is exported to SignInList.js where it is invoked.
+*/
+
+ export const SignInForm = () => {
+     return`
+     <fieldset id="loginForm">
+        <div id="registerButtonContainer"></div>
```
Comment (stevebrownlee) [inline src/scripts/login/SignInForm.js:10]:
```js
${RegisterButton()}
```

**Pull request merged**